### PR TITLE
Fix warnings for iOS 10.3

### PR DESCRIPTION
**Pull request by adempsey**
Also some cleanup

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @adempsey has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m [hunk 1]
```diff
@@ -640,19 +683,32 @@ - (NSURLSessionTask *)backgroundTaskWithRequest:(NSURLRequest *)request orUpload
     return sessionTask;
 }
 
-- (BOOL)persistBackgroundSessionTaskWithSessionId:(NSString *)backgroundSessionId sessionTask:(NSURLSessionTask *)task taskDelegate:(id <BOXURLSessionDownloadTaskDelegate>)taskDelegate userId:(NSString *)userId associateId:(NSString *)associateId error:(NSError **)error
+- (BOOL)persistBackgroundSessionTaskWithSessionId:(NSString *)backgroundSessionId
+                                      sessionTask:(NSURLSessionTask *)task
+                                     taskDelegate:(id <BOXURLSessionTaskDelegate>)taskDelegate
```
Comment (adempsey) [inline BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m:688]: I changed the `taskDelegate` parameter type from `id <BOXURLSessionDownloadTaskDelegate>` to `id <BOXURLSessionTaskDelegate>`. It seems this was intended to use the generic protocol, I don't see any logic specific to downloads here. Let me know if I'm mistaken.

File: BoxContentSDK/BoxContentSDK/AppToApp/BOXAppToAppAnnotationBuilder.m [hunk 2]
```diff
@@ -103,7 +103,7 @@ + (NSString *)stringByDestructivelyParsingInfo:(NSMutableDictionary *)info forKe
 
     if (result != nil)
     {
-        result = [result stringByReplacingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
+        result = [result stringByRemovingPercentEncoding];
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/AppToApp/BOXAppToAppAnnotationBuilder.m:106]: Thanks, the deprecation warnings were driving me crazy!

File: BoxContentSDK/BoxContentSDK/BOXURLRequestSerialization.m [hunk 3]
```diff
@@ -1393,7 +1393,7 @@ - (instancetype)initWithCoder:(NSCoder *)decoder {
         return nil;
     }
 
-    self.format = (NSPropertyListFormat)[[decoder decodeObjectOfClass:[NSNumber class] forKey:NSStringFromSelector(@selector(format))] unsignedIntegerValue];
+    self.format = (NSPropertyListFormat)[[decoder decodeObjectOfClass:[NSNumber class] forKey:NSStringFromSelector(@selector((format)))] unsignedIntegerValue];
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/BOXURLRequestSerialization.m:1396]: Why is this needed? It isn't needed for 'writeOptions' on the next line for example...

File: BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m (hunk 1 above)
Comment (boxtcarpel) [reply BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m:688]: downloads and uploads can be background tasks, others cannot. I'm not sure taskDelegate is the intent here... @thuyBox do you remember?

File: BoxContentSDK/BoxContentSDK/Operations/BOXStreamOperation.m [hunk 4]
```diff
@@ -22,8 +22,6 @@ @interface BOXStreamOperation () <BOXURLSessionDownloadTaskDelegate>
 
 @property (nonatomic, readwrite, assign) unsigned long long bytesReceived;
 
-- (void)writeDataToOutputStream;
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Operations/BOXStreamOperation.m:25]: unimplemented?

File: BoxContentSDK/BoxContentSDK/Operations/BOXStreamOperation.h [hunk 5]
```diff
@@ -96,12 +89,6 @@ typedef void (^BOXAPIStreamProgressBlock)(NSData *data, long long expectedTotalB
 /** @name Overridden methods */
 
 /**
- * In addition to calling [super]([BOXAPIAuthenticatedOperation prepareAPIRequest]), schedule outputStream
- * in the current run loop and set `outputStream.delegate` to `self`.
- */
-- (void)prepareAPIRequest;
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Operations/BOXStreamOperation.h:102]: unimplemented? Or already defined in a superclass?

**Review by boxtcarpel — APPROVED** [review]

File: BoxContentSDK/BoxContentSDK/Operations/BOXStreamOperation.m (hunk 4 above)
Comment (adempsey) [reply BoxContentSDK/BoxContentSDK/Operations/BOXStreamOperation.m:25]: Yes

File: BoxContentSDK/BoxContentSDK/Operations/BOXStreamOperation.h (hunk 5 above)
Comment (adempsey) [reply BoxContentSDK/BoxContentSDK/Operations/BOXStreamOperation.h:102]: Already defined in superclass, it was redeclared here with no actual override.

File: BoxContentSDK/BoxContentSDK/BOXURLRequestSerialization.m (hunk 3 above)
Comment (adempsey) [reply BoxContentSDK/BoxContentSDK/BOXURLRequestSerialization.m:1396]: Several classes expose methods named `format` with different return types, so just using `selector` here is ambiguous as to which method it's actually referring to. But in this case it seems as though we're just using it to get the property name, so I don't think this is causing any problems. The extra layer of parentheses tells the compiler to ignore the warning.

**Review by boxtcarpel — APPROVED** [review]

File: BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m (hunk 1 above)
Comment (thuyBox) [reply BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m:688]: correct, it should have been BOXURLSessionTaskDelegate
